synapse/third_party_rules/access_rules.py

```diff
@@ -25,14 +25,18 @@
 from synapse.types import Requester, get_domain_from_id
 
 ACCESS_RULES_TYPE = "im.vector.room.access_rules"
-ACCESS_RULE_RESTRICTED = "restricted"
-ACCESS_RULE_UNRESTRICTED = "unrestricted"
-ACCESS_RULE_DIRECT = "direct"
+
+
+class AccessRules:
```

Comment:
for future reference, https://docs.python.org/3/library/enum.html might be a useful thing for this sort of construction (saves maintaining a separate VALID_ACCESS_RULES)

Reply:
@richvdh I actually spent about an hour trying to make this work with `Enum` and failed, eventually reverting back to `VALID_ACCESS_RULES`. Reason being that while one can say `isinstance(rule, AccessRule)`, `rule` must be an instance of `AccessRule`. Since we're JSON serialising and deserialising stuff in the tests and the code here, that would've meant having to cast strings to `AccessRule` all over the place.

Thus `VALID_ACCESS_RULES` ended up being the less complex option :)